src/Utils.ts

```diff
@@ -0,0 +1,39 @@
+
+/**
+ * Compares two components.
+ * @remarks
+ *
+ * this recursive function is responsible for comparing two component.
+ *
+ * @param x - the first component that we are going to compare.
+ * @param y - the second component that we are going to compare.
+ * @returns true, if both components are identical; false, if the components are not identical.
```

Comment:
You should add the types for this JSDoc params and return

Reply:
@magicmatatjahu I am using TSDOC to annotate and generate documentation. It will figure out the types in the documentation and in code, well, it is in the function definition. If you think TSDOC is not mature enough, I will change them to JSDOC.

Reply:
TBH I have no opinion on this. In the other repos, we use the JSDoc, so others would suggest you to use the JSDoc... probably JSDoc will be better now :)